src/extra/Form/mermaid.tsx

```diff
@@ -41,64 +22,49 @@ class Widget extends React.Component<FormWidgetProps, WidgetState> {
 			...inputProps
 		} = this.props;
 
-		const { showPreview } = this.state;
-
 		const change = ({ target: { value } }: any) => {
 			return this.props.onChange(value === '' ? options.emptyValue : value);
 		};
 
 		return (
-			<Box>
-				<Flex justify="space-between" mb={1}>
-					<Flex>
-						<Button plaintext mr={3} onClick={this.hidePreview}>
-							<Txt bold={!showPreview}>Write</Txt>
-						</Button>
-
-						<Button plaintext onClick={this.showPreview}>
-							<Txt bold={showPreview}>Preview</Txt>
-						</Button>
-					</Flex>
-
+			<TabbedBox
+				tabs={['Write', 'Preview']}
+				help={
 					<Link fontSize={2} href={'https://mermaidjs.github.io/'} blank>
 						About mermaid
 					</Link>
-				</Flex>
-
-				{showPreview && (
-					<Mermaid
-						bg={Theme.colors.gray.light}
-						p={3}
-						style={{
-							borderRadius: Theme.radius,
-							width: '100%',
-							minHeight: 180,
-						}}
-						value={value}
-					/>
-				)}
+				}
+			>
+				<Textarea
+					placeholder="This text supports mermaid charts"
+					readOnly={readonly}
+					disabled={disabled}
+					autoFocus={autofocus}
+					value={value || ''}
+					{...inputProps}
+					onChange={change}
+					onBlur={
+						onBlur &&
+						((event: any) => onBlur(inputProps.id, event.target.value))
+					}
+					onFocus={
+						onFocus &&
+						((event: any) => onFocus(inputProps.id, event.target.value))
```

Comment:
We could easily change these as well but definitely not mandatory atm.

Reply:
I'll change them in this PR